src/DGtal/geometry/curves/estimation/FunctorsLambdaMST.h

```diff
@@ -0,0 +1,218 @@
+#include "DGtal/geometry/curves/estimation/CLambdaFunctor.h"
+//////////////////////////////////////////////////////////////////////////////
+
+namespace DGtal
+{
+/////////////////////////////////////////////////////////////////////////////
+/**
+  * Description: Extension of namespace functors by functors related to L-MST.
+  * \brief Aim: Provide various lambda functions and others L-MST related functors.
+  * 
+*/
+
+namespace functors
+{
+  /**
+   * Model of CLambdaFunctor -- polynomial functor
+   * 
+   */
+  struct Lambda64Function : std::unary_function < double, double >
+  {
+      double operator() (double x) const
+      {
+          double e2 = x * x;
+          double e3 = e2 * x;
+          return 64.0 * ( -e3 * e3 + 3.0 * e3 * e2 - 3.0 * e2 * e2 + e3 );
+      }
+  };
+  /**
+   * Model of CLambdaFunctor -- sine lambda-functor
+   * 
+   */
+  struct LambdaSinFromPiFunction : std::unary_function < double, double >
+  {
+      double operator() (double x) const
+      {
+          return std::sin ( M_PI * x );
+      }
+  };
+
+  /**
+   * Model of CLambdaFunctor -- exponential lambda-functor
+   * 
+   */
+  struct LambdaExponentialFunction : std::unary_function < double, double >
+  {
+      double operator() (double x) const
+      {
+          return 2.0 / ( std::exp ( 15.0 * ( x - 0.5 ) ) + std::exp ( -15.0 * ( x - 0.5 ) ) );
+      }
+  };
+}
+
+/**
+ * Description of class 'TangentFromDSS2DFunctor' -- model of CLMSTTangentFromDSS.
+ * Aim: Provide a functor which calculate from digital straight segment 
+ * its direction and eccentricity around a given point.
+ * @tparam DSS digital straight segment recognition algorithm
+ * @tparam LambdaFunction model of CLambdaFunctor @see CLambdaFunctor.h
+ */
+template<typename DSS, typename LambdaFunction>
+class TangentFromDSS2DFunctor
+{
+  // ----------------------- Types ------------------------------
+public:
+  typedef PointVector<2, double> RealVector;
+  typedef DSS TDSS;
+  
+  struct Value
+  {
+    RealVector first;
+    double second;
+    Value () : second ( 0. ) {}
+    Value & operator+= ( const Value & ch )
+    {
+      this->first += ch.first;
+      this->second += ch.second;
+      return *this;
+    }
+  };
+  
+  
+  // ----------------------- Interface --------------------------------------
+public:
+  /**
+   * Calculate a direction of the 2D DSS and an eccentricity of a given point in
+   * this DSS.
+   * @param aDSS digital straight segment
+   * @param indexOfPointInDSS index of given point in aDSS
+   * @param dssLen length of aDSS
+   */
+  Value operator() ( const TDSS& aDSS, const int & indexOfPointInDSS, const int & dssLen ) const
+  {
+    Value result;
+    double norm = std::sqrt ( aDSS.a() * aDSS.a() + aDSS.b() * aDSS.b() );
+    result.second = lambdaFunctor ( (double)indexOfPointInDSS / (double)dssLen );
+    result.first[0] = result.second * aDSS.a () / norm;
+    result.first[1] = result.second * aDSS.b () / norm;
```

Comment:
Yes there is a Fraction class. Otherwise, you could simply output the pair (a,b).

BTW, since there is a functor (on double) to filter from  eccentricity  and tangents in the pencil, Maybe exact output does not make sense (expect of the filtering kernel is integer based)...